### PR TITLE
test/e2e: improve waits, logs, etc

### PR DESCRIPTION
**Pull request by stevekuznetsov**

### CODE REVIEW

File: deploy/chart/templates/0000_50_olm_07-olm-operator.deployment.yaml [hunk 1]
```diff
@@ -110,6 +110,10 @@ spec:
                 fieldPath: metadata.namespace
           - name: OPERATOR_NAME
             value: olm-operator
+          {{- if .Values.debug }}
+          - name: CI
+            value: "true"
+          {{- end }}
```
[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply deploy/chart/templates/0000_50_olm_07-olm-operator.deployment.yaml:116]: /shrug

The current workflow in this repo is to evaluate this helm chart with the `.Values.debug` set. A fail-fast feature that I added should run in CI, where `$CI` is set. I'm just making a minimal update to map A to B.

[the remark this answers is not in the document]
Comment (stevekuznetsov) [reply deploy/chart/templates/0000_50_olm_07-olm-operator.deployment.yaml:116]: > It just feels weird

Amen!

**Review by ncdc — APPROVED** [review]

File: test/e2e/crd_e2e_test.go [hunk 2]
```diff
@@ -261,8 +261,9 @@ var _ = Describe("CRD Versions", func() {
 		})).Should(Succeed())
 		ctx.Ctx().Logf("updated subscription to point to alpha channel")
 
+		checker := subscriptionStateAtLatestChecker()
 		subscriptionAtLatestWithDifferentInstallPlan := func(v *operatorsv1alpha1.Subscription) bool {
-			return subscriptionStateAtLatestChecker(v) && v.Status.InstallPlanRef != nil && v.Status.InstallPlanRef.Name != fetchedInstallPlan.Name
+			return checker(v) && v.Status.InstallPlanRef != nil && v.Status.InstallPlanRef.Name != fetchedInstallPlan.Name
```
Comment (tmshort) [inline test/e2e/crd_e2e_test.go:266]: `checker` seems redundant? What's this change for?

File: test/e2e/util.go [hunk 3]
```diff
@@ -337,19 +337,27 @@ func registryPodHealthy(address string) bool {
...
-	if registry != nil && connState != nil && !connState.LastConnectTime.IsZero() && connState.LastObservedState == "READY" {
-		fmt.Printf("probing catalog %s pod with address %s\n", catalog.GetName(), registry.Address())
-		return registryPodHealthy(registry.Address())
-	}
-	state := "NO_CONNECTION"
-	if connState != nil {
-		state = connState.LastObservedState
+func catalogSourceRegistryPodSynced() func(catalog *operatorsv1alpha1.CatalogSource) bool {
+	var lastState string
+	lastTime := time.Now()
```
Comment (tmshort) [inline test/e2e/util.go:342]: Interesting... (more of a "oh, wait, golang let's you do some interesting stuff")

**Review by tmshort — APPROVED** [review]
Just a question and an observation...

File: test/e2e/crd_e2e_test.go (hunk 2 above)
Comment (stevekuznetsov) [reply test/e2e/crd_e2e_test.go:266]: I changed the signature - it's got one more level of nesting to the closure now.

Comment (stevekuznetsov) [reply test/e2e/crd_e2e_test.go:266]: So `checker(v)` is `subscriptionStateAtLatestChecker()(v)`

**Comment by tmshort** [comment]
Tests need to be redone, but I'm not kicking off resting in case @stevekuznetsov wants to use the output...

**Comment by stevekuznetsov** [comment]
Can't make heads or tails of those failures or how I possibly broke them .... :facepalm:

**Pull request merged**